client/app/visualizations/counter/utils.js

```diff
@@ -70,31 +70,27 @@ function formatTooltip(value, formatString) {
   return toString(value);
 }
 
-export function getCounterData(rows, options, visualizationName) {
+export function getCounterData(rows = [], options = {}, visualizationName = '') {
```

Comment:
I agree refactoring code base for the sole sake of testing is generally not ideal.

`rows = []` was meant to protect against calling `getCounterData()` at runtime which would trow TypeError on `rows.length` (and since there is no type checking at build time). Fair enough such case is not realistic and `rows` should be a mandatory param.

Same for `options = {}` although the nature of options may suggest they should be _optional_. In that sense keeping the default value is not just for tests and the same is true for `visualizationName = ''` which one may want to fallback to an empty string.

Would following amended function signature do the job here?:
`function getCounterData(rows, options = {}, visualizationName = '')`

Reply:
There will always be options and a visualization name... I'm actually not sure about `rows` as I don't know what happens when we're still waiting for the results to load. But having a default value won't help in such case anyway, because we definitely pass some value, it just might be empty.

My suggestion is to revert the function definition to be as it was to make it clear that it needs these values, but for convenience in the test you can wrap it with a function that has default values.

Reply:
ok cool, default param values removed